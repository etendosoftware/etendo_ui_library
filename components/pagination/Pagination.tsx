/* Imports */
import React, {useState} from 'react';

import {Text, View, TouchableOpacity} from 'react-native';
import {PaginationStyleVariant} from './Pagination.styles';
import {PaginationProps} from './Pagination.types';
import {
  arrowFirst,
  arrowLast,
  arrowPickerHorizontalIcon,
} from '../../assets/images/icons';
import CustomImage from '../../SecondaryComponents/CustomImage';

/* Pagination component */
export const Pagination = ({
  currentPage,
  onChangeSelected,
  totalData,
  amountDataPerPage,
  pagination,
}: PaginationProps) => {
  // Will serve to set the page number where the user is
  const [page, setPage] = useState<number>(currentPage);

  // This constant will count the number of pages that the data table will have based on the number of row
  // that the progammer wishes to display per page
  const pageNumbers: Array<number> = [];
  for (let i = 1; i <= Math.ceil(totalData / amountDataPerPage); i++) {
    pageNumbers.push(i);
  }

  /* Helper functions to page up or down */
  const backToFirstPage = () => {
    setPage(1);
  };

  const backPage = (page: number) => {
    setPage(page - 1);
  };

  const nextPage = (page: number) => {
    setPage(page + 1);
  };

  const nextLastPage = (pageNumbers: Array<number>) => {
    setPage(pageNumbers.length);
  };

  return (
    <View style={PaginationStyleVariant.primary.container}>
      <View style={PaginationStyleVariant.primary.container}>
        <TouchableOpacity
          disabled={page <= 1 || totalData === 0}
          onPress={() => {
            /* Allows to detect the selected item */
            backToFirstPage();
            onChangeSelected(1, 0);
          }}
        >
          <CustomImage
            src={arrowFirst}
            alt="back-double-button"
            source={{uri: arrowFirst}}
            style={{
              width: 10,
              height: 10,
              marginRight: 10,
            }}
          />
        </TouchableOpacity>

        <TouchableOpacity
          disabled={page <= 1 || totalData === 0}
          onPress={() => {
            backPage(page);
            if (page !== 1) {
              onChangeSelected(page - 1, pagination - amountDataPerPage);
            }
          }}
        >
          <CustomImage
            src={arrowPickerHorizontalIcon}
            alt="back-button"
            source={{uri: arrowPickerHorizontalIcon}}
            style={{
              width: 10,
              height: 10,
              marginRight: 10,
              rotate: '180deg',
            }}
          />
        </TouchableOpacity>

        {/* Text with layout to display the page number the user is on */}
        <View style={PaginationStyleVariant.primary.pageNumberContainer}>
          <Text style={PaginationStyleVariant.primary.pageNumberText}>
            {page}
          </Text>
        </View>

        <TouchableOpacity
          disabled={page === pageNumbers.length || totalData === 0}
          onPress={() => {
            nextPage(page);
            if (page !== Math.ceil(totalData / 8)) {
              onChangeSelected(page + 1, pagination + amountDataPerPage);
            }
          }}
        >
          <CustomImage
            style={{
              width: 10,
              height: 10,
              marginRight: 10,
            }}
            src={arrowPickerHorizontalIcon}
            alt="next-button"
            source={{uri: arrowPickerHorizontalIcon}}
          />
        </TouchableOpacity>

        <TouchableOpacity
          disabled={page === pageNumbers.length || totalData === 0}
          onPress={() => {
            nextLastPage(pageNumbers);
            if (page !== Math.ceil(totalData / 8)) {
              onChangeSelected(
                Math.ceil(totalData / amountDataPerPage),
                totalData - (totalData % amountDataPerPage) - 8,
              );
            }
          }}
        >
          <CustomImage
            style={{
              width: 10,
              height: 10,
              marginRight: 10,
            }}
            src={arrowLast}
            alt="next-double-button"
            source={{uri: arrowLast}}
          />
        </TouchableOpacity>
      </View>
    </View>
  );
};
