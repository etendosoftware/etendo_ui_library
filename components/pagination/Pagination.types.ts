/* Imports */
import {ImageStyle, StyleProp, TextStyle, ViewStyle} from 'react-native';

/* Type declaration */
export type Info = {
  name: string;
  route: string;
  key: string;
};

export interface PaginationProps {
  currentPage: number;
  onChangeSelected: (Info: Info | undefined) => Info | undefined;
  totalData: number;
  amountDataPerPage: number;
}

/* Declaration of style types */
export type TabStyleType = 'primary';

export type PaginationStyleRecord = Record<
  TabStyleType,
  {
    container: ViewStyle;
    icon?: ImageStyle;
    pageNumberContainer: ViewStyle;
    pageNumberText: TextStyle;
  }
>;
