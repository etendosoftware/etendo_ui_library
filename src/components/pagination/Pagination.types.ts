/* Imports */
import {ImageStyle, TextStyle, ViewStyle} from 'react-native';

/* Type declaration */
export type Info = {
  name: string;
  route: string;
  key: string;
};

export interface PaginationProps {
  currentPage: number;
  onChangeSelected: ({type}: any, {type2}: any) => void;
  totalData: number;
  amountDataPerPage: number;
  pagination: number;
  isWeb?: boolean;
}

/* Declaration of style types */
export type TabStyleType = 'primary';

export type PaginationStyleRecord = Record<
  TabStyleType,
  {
    container: ViewStyle;
    icon?: ImageStyle;
    pageNumberContainer: ViewStyle;
    pageNumberText: TextStyle;
  }
>;
