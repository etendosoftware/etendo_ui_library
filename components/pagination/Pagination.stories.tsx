import React from 'react';
import {Story, Meta} from '@storybook/react';

import {Pagination} from './Pagination';
import {PaginationProps} from './Pagination.types';

export default {
  title: 'Etendo/Pagination',
  component: Pagination,
  argTypes: {},
} as Meta<typeof Pagination>;

/* Information to pass */
const dataset = [
  {name: 'User', route: '', key: 'user'},
  {name: 'Vehicles', route: '', key: 'vehicles'},
  {name: 'Skills', route: '', key: 'skills'},
  {name: 'Mobile app', route: '', key: 'movilApp'},
  {name: 'Staff', route: '', key: 'staff'},
  {name: 'Routing options', route: '', key: 'routing options'},
  {name: 'Map', route: '', key: 'map'},
];

/* This function allows you to select by default the first tab */
const onCheckSelectedMock = () => dataset[0];

/* Templates */
const Template: Story<PaginationProps> = args => (
  <Pagination
    currentPage={1}
    onChangeSelected={onCheckSelectedMock}
    totalData={40}
    amountDataPerPage={8}
  />
);

/* Story exports */
export const PaginationDefault = Template.bind({});
PaginationDefault.args = {};