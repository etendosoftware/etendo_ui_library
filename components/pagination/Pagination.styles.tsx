/* Imports */
import {PaginationStyleRecord} from './Pagination.types';
import {PURPLE_50} from '../../styles/colors';
import { StyleSheet } from 'react-native';

/* Export of different styles */
export const PaginationStyleVariant: PaginationStyleRecord = {
  /* Primary style */
  primary: {
    container: {
      flex: 1,
      width: 120,
      height: 22,
      flexDirection: 'row',
      justifyContent: 'flex-end',
      alignItems: 'center',
    },
    icon: {
      width: 22,
      height: 22,
      marginRight: 17,
    },
    pageNumberContainer: {
      backgroundColor: PURPLE_50,
      width: 25,
      height: 25,
      borderRadius: 3,
      justifyContent: 'center',
      alignSelf: 'center',
      marginRight: 12,
    },
    pageNumberText: {
      fontWeight: '500',
      fontSize: 12,
      alignSelf: 'center',
    },
  },
};

export const styles = StyleSheet.create({
  storiesContainer: {
    height: 50,
  }
});
