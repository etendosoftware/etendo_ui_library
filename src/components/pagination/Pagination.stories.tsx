import React from 'react';
import {Story, Meta} from '@storybook/react';

import {Pagination} from './Pagination';
import {PaginationProps} from './Pagination.types';
import {View} from 'react-native';
import addMarginContainer from '../../helpers/addMargin';
import { styles } from './Pagination.styles';

export default {
  title: 'Etendo/Pagination',
  component: Pagination,
  argTypes: {},
} as Meta<typeof Pagination>;

/* Information to pass */
const dataset = [
  {name: 'User', route: '', key: 'user'},
  {name: 'Vehicles', route: '', key: 'vehicles'},
  {name: 'Skills', route: '', key: 'skills'},
  {name: 'Mobile app', route: '', key: 'movilApp'},
  {name: 'Staff', route: '', key: 'staff'},
  {name: 'Routing options', route: '', key: 'routing options'},
  {name: 'Map', route: '', key: 'map'},
];

/* This function allows you to select by default the first tab */
const onCheckSelectedMock = () => dataset[0];

/* Templates */
const Template: Story<PaginationProps> = args => (
  <View style={[styles.storiesContainer,addMarginContainer()]}>
    <Pagination
      currentPage={1}
      onChangeSelected={onCheckSelectedMock}
      totalData={40}
      amountDataPerPage={8}
      pagination={0}
    />
  </View>
);

/* Story exports */
export const PaginationDefault = Template.bind({});
PaginationDefault.args = {};
