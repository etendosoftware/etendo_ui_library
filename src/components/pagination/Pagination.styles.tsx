import { QUATERNARY_50 } from '../../styles/colors';
import { PaginationStyleVariant } from './Pagination.types';
import { StyleSheet } from 'react-native';

export const paginationStyleVariant: PaginationStyleVariant = {
  primary: {
    container: {
      flex: 1,
      width: 120,
      height: 22,
      flexDirection: 'row',
      alignItems: 'center',
    },
    icon: {
      width: 22,
      height: 22,
    },
    pageNumberContainer: {
      backgroundColor: QUATERNARY_50,
      width: 25,
      height: 25,
      borderRadius: 3,
      justifyContent: 'center',
      alignSelf: 'center',
      marginHorizontal: 5,
    },
    pageNumberText: {
      fontWeight: '500',
      fontSize: 12,
      alignSelf: 'center',
    },
  },
};

export const styles = StyleSheet.create({
  storiesContainer: {
    height: 50,
  },
});
