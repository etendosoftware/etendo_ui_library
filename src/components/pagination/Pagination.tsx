/* Imports */
import React, { useState } from 'react';

import { View, TouchableOpacity, Text } from 'react-native';
import { PaginationStyleVariant } from './Pagination.styles';
import { PaginationProps } from './Pagination.types';
import { ArrowLeftIcon } from '../../assets/images/icons/ArrowLeftIcon';
import { ArrowFirstIcon } from '../../assets/images/icons/ArrowFirstIcon';
import { ArrowRightIcon } from '../../assets/images/icons/ArrowRightIcon';
import { ArrowLastIcon } from '../../assets/images/icons/ArrowLastIcon';

/* Pagination component */
const Pagination = ({
  currentPage,
  onChangeSelected,
  totalData,
  amountDataPerPage,
  pagination,
}: PaginationProps) => {
  // Will serve to set the page number where the user is
  const [page, setPage] = useState<number>(currentPage);

  // This constant will count the number of pages that the data table will have based on the number of row
  // that the progammer wishes to display per page
  const pageNumbers: Array<number> = [];
  for (let i = 1; i <= Math.ceil(totalData / amountDataPerPage); i++) {
    pageNumbers.push(i);
  }

  /* Helper functions to page up or down */
  const backToFirstPage = () => {
    setPage(1);
  };

  const backPage = (page: number) => {
    setPage(page - 1);
  };

  const nextPage = (page: number) => {
    setPage(page + 1);
  };

  const nextLastPage = (pageNumbers: Array<number>) => {
    setPage(pageNumbers.length);
  };
  return (
    <View style={PaginationStyleVariant.primary.container}>
      <View style={PaginationStyleVariant.primary.container}>
        <TouchableOpacity
          disabled={page <= 1 || totalData === 0}
          onPress={() => {
            /* Allows to detect the selected item */
            backToFirstPage();
            onChangeSelected(1, 0);
          }}>
          <ArrowFirstIcon
            style={{
              width: 10,
              height: 10,
              marginRight: 10,
            }}
          />
        </TouchableOpacity>

        <TouchableOpacity
          disabled={page <= 1 || totalData === 0}
          onPress={() => {
            backPage(page);
            if (page !== 1) {
              onChangeSelected(page - 1, pagination - amountDataPerPage);
            }
          }}>
          <ArrowLeftIcon
            style={{
              height: 10,
              width: 10,
              marginRight: 10,
            }}
          />
        </TouchableOpacity>

        {/* Text with layout to display the page number the user is on */}
        <View style={PaginationStyleVariant.primary.pageNumberContainer}>
          <Text style={PaginationStyleVariant.primary.pageNumberText}>
            {page}
          </Text>
        </View>

        <TouchableOpacity
          disabled={page === pageNumbers.length || totalData === 0}
          onPress={() => {
            nextPage(page);
            if (page !== Math.ceil(totalData / 8)) {
              onChangeSelected(page + 1, pagination + amountDataPerPage);
            }
          }}>
          <ArrowRightIcon
            style={{
              height: 10,
              width: 10,
              marginRight: 10,
            }}
          />
        </TouchableOpacity>

        <TouchableOpacity
          disabled={page === pageNumbers.length || totalData === 0}
          onPress={() => {
            nextLastPage(pageNumbers);
            if (page !== Math.ceil(totalData / 8)) {
              onChangeSelected(
                Math.ceil(totalData / amountDataPerPage),
                totalData - (totalData % amountDataPerPage) - 8,
              );
            }
          }}>
          <ArrowLastIcon
            style={{
              height: 10,
              width: 10,
              marginRight: 10,
            }}
          />
        </TouchableOpacity>
      </View>
    </View>
  );
};

export default Pagination;
