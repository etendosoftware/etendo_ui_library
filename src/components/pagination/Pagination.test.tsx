import React from 'react';
import renderer from 'react-test-renderer';

import { Pagination } from '../pagination/Pagination';

const dataset = [
  { name: 'User', route: '', key: 'user' },
  { name: 'Vehicles', route: '', key: 'vehicles' },
  { name: 'Skills', route: '', key: 'skills' },
  { name: 'Mobile app', route: '', key: 'movilApp' },
  { name: 'Staff', route: '', key: 'staff' },
  { name: 'Routing options', route: '', key: 'routing options' },
  { name: 'Map', route: '', key: 'map' },
];

const onCheckSelectedMock = () => dataset[0];

describe('Running Test for Pagination', () => {
  it('Check Pagination Disabled', () => {
    const tree = renderer
      .create(
        <Pagination
          currentPage={1}
          onChangeSelected={onCheckSelectedMock}
          totalData={40}
          amountDataPerPage={8}
          pagination={0}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
