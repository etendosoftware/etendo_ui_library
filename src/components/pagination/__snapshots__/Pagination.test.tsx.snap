// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Running Test for Pagination Check Pagination Disabled 1`] = `
<div
  className="css-view-175oi2r"
  dir={null}
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "row",
      "height": "22px",
      "justifyContent": "flex-end",
      "width": "120px",
    }
  }
>
  <div
    className="css-view-175oi2r"
    dir={null}
    style={
      {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "height": "22px",
        "justifyContent": "flex-end",
        "width": "120px",
      }
    }
  >
    <div
      aria-disabled={true}
      className="css-view-175oi2r r-transitionProperty-1i6wzkk r-userSelect-lrvibr r-pointerEvents-633pao"
      dir={null}
      onClick={[Function]}
      onContextMenu={[Function]}
      onKeyDown={[Function]}
      style={
        {
          "transitionDuration": "0s",
        }
      }
      tabIndex="-1"
    >
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight="10"
        bbWidth="10"
        fill="none"
        focusable={false}
        height={10}
        meetOrSlice={0}
        minX={0}
        minY={0}
        style={
          [
            {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            {
              "height": 10,
              "marginRight": 10,
              "width": 10,
            },
            {
              "flex": 0,
              "height": 10,
              "width": 10,
            },
          ]
        }
        vbHeight={32}
        vbWidth={31}
        width={10}
      >
        <RNSVGGroup
          fill={null}
          propList={
            [
              "fill",
            ]
          }
        >
          <RNSVGPath
            clipRule={0}
            d="M1.7901 0C2.16123 3.24449e-08 2.53235 6.48898e-08 2.9034 9.73277e-08C3.18664 0.113646 3.49781 0.184106 3.74726 0.349623C4.29665 0.714262 4.51729 1.26406 4.51632 1.92727L4.52659 29.9793C4.53422 31.2575 3.6186 32.0643 2.52882 31.996C2.43644 31.9902 2.34342 31.9956 2.25064 31.9956C0.999848 31.9957 0.310647 31.3086 0.310487 30.0521C0.309605 20.691 0.309766 11.3299 0.310409 1.96874C0.310489 0.814188 0.711072 0.35571 1.7901 0ZM8.55658 12.8884C14.1996 8.9659 19.8425 5.04343 25.4815 1.11527C26.1889 0.622537 26.9056 0.16357 27.7671 2.27098e-06C27.9733 2.28901e-06 28.1794 2.30703e-06 28.3856 2.32505e-06C29.3666 0.206594 29.9889 0.820523 30.2563 1.77401C30.3866 2.2389 30.4741 2.73374 30.4748 3.21528C30.4876 11.733 30.4628 20.2507 30.4307 28.7685C30.4282 29.3611 30.3978 29.9906 30.0936 30.5684C29.7533 31.5261 28.8343 32.005 27.8727 31.9526C27.1081 31.8723 26.4613 31.5231 25.8459 31.0956C20.0833 27.0919 14.3211 23.0875 8.56204 19.0785C8.25801 18.8668 7.95799 18.637 7.70028 18.3718C6.72165 17.3098 6.3673 16.1896 6.87479 14.8316C7.19303 13.98 7.8311 13.3809 8.55658 12.8884Z"
            fill={
              {
                "payload": 4280296530,
                "type": 0,
              }
            }
            fillRule={0}
            propList={
              [
                "fill",
                "fillRule",
              ]
            }
          />
        </RNSVGGroup>
      </RNSVGSvgView>
    </div>
    <div
      aria-disabled={true}
      className="css-view-175oi2r r-transitionProperty-1i6wzkk r-userSelect-lrvibr r-pointerEvents-633pao"
      dir={null}
      onClick={[Function]}
      onContextMenu={[Function]}
      onKeyDown={[Function]}
      style={
        {
          "transitionDuration": "0s",
        }
      }
      tabIndex="-1"
    >
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight="10"
        bbWidth="10"
        fill="none"
        focusable={false}
        height={10}
        meetOrSlice={0}
        minX={0}
        minY={0}
        style={
          [
            {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            {
              "height": 10,
              "marginRight": 10,
              "width": 10,
            },
            {
              "flex": 0,
              "height": 10,
              "width": 10,
            },
          ]
        }
        vbHeight={32}
        vbWidth={24}
        width={10}
      >
        <RNSVGGroup
          fill={null}
          propList={
            [
              "fill",
            ]
          }
        >
          <RNSVGPath
            clipRule={0}
            d="M2.28888 12.5093C0.099184 14.3213 0.0991821 17.6787 2.28888 19.4907L16.1377 30.9511C19.0917 33.3956 23.5574 31.2947 23.5574 27.4604L23.5574 4.5396C23.5574 0.705348 19.0917 -1.39562 16.1377 1.04888L2.28888 12.5093Z"
            fill={
              {
                "payload": 4280296530,
                "type": 0,
              }
            }
            fillRule={0}
            propList={
              [
                "fill",
                "fillRule",
              ]
            }
          />
        </RNSVGGroup>
      </RNSVGSvgView>
    </div>
    <div
      className="css-view-175oi2r"
      dir={null}
      style={
        {
          "alignSelf": "center",
          "backgroundColor": "rgba(221,223,246,1.00)",
          "borderBottomLeftRadius": "3px",
          "borderBottomRightRadius": "3px",
          "borderTopLeftRadius": "3px",
          "borderTopRightRadius": "3px",
          "height": "25px",
          "justifyContent": "center",
          "marginRight": "12px",
          "width": "25px",
        }
      }
    >
      <div
        className="css-text-146c3p1"
        dir="auto"
        style={
          {
            "alignSelf": "center",
            "fontSize": "12px",
            "fontWeight": "500",
          }
        }
      >
        1
      </div>
    </div>
    <div
      className="css-view-175oi2r r-transitionProperty-1i6wzkk r-userSelect-lrvibr r-cursor-1loqt21 r-touchAction-1otgn73"
      dir={null}
      onClick={[Function]}
      onContextMenu={[Function]}
      onKeyDown={[Function]}
      style={
        {
          "transitionDuration": "0s",
        }
      }
      tabIndex="0"
    >
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight="10"
        bbWidth="10"
        fill="none"
        focusable={false}
        height={10}
        meetOrSlice={0}
        minX={0}
        minY={0}
        style={
          [
            {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            {
              "height": 10,
              "marginRight": 10,
              "width": 10,
            },
            {
              "flex": 0,
              "height": 10,
              "width": 10,
            },
          ]
        }
        vbHeight={32}
        vbWidth={23}
        width={10}
      >
        <RNSVGGroup
          fill={null}
          propList={
            [
              "fill",
            ]
          }
        >
          <RNSVGPath
            d="M21.2914 12.5093C23.4811 14.3213 23.4811 17.6787 21.2914 19.4907L7.44262 30.9511C4.48866 33.3956 0.0229495 31.2947 0.0229495 27.4604L0.0229492 4.5396C0.0229492 0.705348 4.48865 -1.39562 7.44261 1.04888L21.2914 12.5093Z"
            fill={
              {
                "payload": 4280296530,
                "type": 0,
              }
            }
            propList={
              [
                "fill",
              ]
            }
          />
        </RNSVGGroup>
      </RNSVGSvgView>
    </div>
    <div
      className="css-view-175oi2r r-transitionProperty-1i6wzkk r-userSelect-lrvibr r-cursor-1loqt21 r-touchAction-1otgn73"
      dir={null}
      onClick={[Function]}
      onContextMenu={[Function]}
      onKeyDown={[Function]}
      style={
        {
          "transitionDuration": "0s",
        }
      }
      tabIndex="0"
    >
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight="10"
        bbWidth="10"
        fill="none"
        focusable={false}
        height={10}
        meetOrSlice={0}
        minX={0}
        minY={0}
        style={
          [
            {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            {
              "height": 10,
              "marginRight": 10,
              "width": 10,
            },
            {
              "flex": 0,
              "height": 10,
              "width": 10,
            },
          ]
        }
        vbHeight={32}
        vbWidth={31}
        width={10}
      >
        <RNSVGGroup
          fill={null}
          propList={
            [
              "fill",
            ]
          }
        >
          <RNSVGPath
            clipRule={0}
            d="M29.1664 32C28.7953 32 28.4241 32 28.0531 32C27.7698 31.8864 27.4587 31.8159 27.2092 31.6504C26.6598 31.2857 26.4392 30.7359 26.4402 30.0727L26.4299 2.02074C26.4223 0.742552 27.3379 -0.0642497 28.4277 0.00401875C28.52 0.0097822 28.6131 0.00442463 28.7058 0.00442463C29.9566 0.00434345 30.6458 0.691411 30.646 1.94792C30.6469 11.309 30.6467 20.6701 30.6461 30.0313C30.646 31.1858 30.2454 31.6443 29.1664 32ZM22.3999 19.1116C16.7569 23.0341 11.114 26.9566 5.47496 30.8847C4.76762 31.3775 4.05089 31.8364 3.18937 32C2.98318 32 2.77707 32 2.57088 32C1.58993 31.7934 0.967588 31.1795 0.700239 30.226C0.569895 29.7611 0.482411 29.2663 0.481688 28.7847C0.468846 20.267 0.493727 11.7493 0.525832 3.23155C0.52824 2.63889 0.558739 2.00937 0.862928 1.43157C1.20323 0.473862 2.12222 -0.00499171 3.08383 0.0473664C3.84839 0.12773 4.49522 0.476865 5.11058 0.904415C10.8732 4.90806 16.6354 8.91252 22.3944 12.9215C22.6985 13.1332 22.9985 13.363 23.2562 13.6282C24.2348 14.6902 24.5892 15.8104 24.0817 17.1684C23.7635 18.02 23.1254 18.6191 22.3999 19.1116Z"
            fill={
              {
                "payload": 4280296530,
                "type": 0,
              }
            }
            fillRule={0}
            propList={
              [
                "fill",
                "fillRule",
              ]
            }
          />
        </RNSVGGroup>
      </RNSVGSvgView>
    </div>
  </div>
</div>
`;
